code/modules/reagents/Chemistry-Holder.dm

```diff
@@ -44,19 +44,47 @@ var/const/INGEST = 2
 
 		chemical_reactions_list = list()
 
-		for (var/path in typesof(/datum/chemical_reaction) - /datum/chemical_reaction)
+		//variables that we want to reuse
+		var/list/reaction_ids = list()
+		var/smallest_number_of_reactants = INFINITY
+		var/smallest_reactants_list_index = 1
+		var/list/reactant_list
+		var/datum/chemical_reaction/D
+		var/i
 
-			var/datum/chemical_reaction/D = new path()
-			var/list/reaction_ids = list()
+		for(var/path in typesof(/datum/chemical_reaction) - /datum/chemical_reaction)
+
+			D = new path()
+			reaction_ids.len = 0
 
 			if(D.required_reagents && D.required_reagents.len)
-				var/reaction = D.required_reagents[1]
-				if(islist(reaction))
-					var/list/L = reaction
-					for(var/content in L)
-						reaction_ids += content
-				else
-					reaction_ids += reaction
+
+				//to minimize the size of the reactions lists, we ideally want each reaction that requires an individual (non-list) reagent to have that as the "key" reagent of the reaction
+				//if a reaction only requires lists of reagents, then we want to pick the smallest list
+				smallest_number_of_reactants = INFINITY
+				smallest_reactants_list_index = 1
+
+				i = 0
+				for(var/reactant in D.required_reagents)
+					i++
+					if(islist(reactant))
+						reactant_list = reactant
+						if(smallest_reactants_list_index)
```

Comment:
since this was just defined I think it would always be true, right?

Reply:
ty